### PR TITLE
[202211][db_migrator] Add migration of FLEX_COUNTER_DELAY_STATUS during 1911->2211 upgrade + fast-reboot. Add UT.

### PR DESCRIPTION
**Pull request by vadymhlushko-mlnx**
#### What I did
Add migration of `FLEX_COUNTER_DELAY_STATUS` attribute of config_db `FLEX_COUNTER_TABLE` during the `SONiC to SONiC upgrade + fast-reboot` from older versions `201911 -> 202211`.

This change is required for the `fast-reboot` procedure because without it the counters will be created during the init flow which will waste a lot of resources and cause data plane degradation of more than 30 seconds.

#### How I did it
Modify the `db_migrator.py`.

#### How to verify it
Add UT.

### CODE REVIEW

**Review by stepanblyschak — APPROVED** [review]

**Review by stepanblyschak — APPROVED** [review]

**Comment by keboliu** [comment]
@vaibhavhd  would you please help to review?

File: scripts/db_migrator.py [hunk 1]
```diff
@@ -884,6 +898,10 @@ def version_3_0_6(self):
         """
 
         log.log_info('Handling version_3_0_6')
+
+        if self.stateDB.keys(self.stateDB.STATE_DB, "FAST_REBOOT|system"):
+            self.migrate_config_db_flex_counter_delay_status()
```
[the remark this answers is not in the document]
Comment (vadymhlushko-mlnx) [reply scripts/db_migrator.py:902]: Resolved

**Comment by vaibhavhd** [comment]
Why is this change not needed on master?

**Review by vaibhavhd — APPROVED** [review]

**Comment by keboliu** [comment]
@vadymhlushko-mlnx please resolve the checker issue so we can proceed.

**Comment by vadymhlushko-mlnx** [comment]
> Why is this change not needed on master?

The PR's for other branches have already been created:

https://github.com/sonic-net/sonic-utilities/pull/2839
https://github.com/sonic-net/sonic-utilities/pull/2870
https://github.com/sonic-net/sonic-utilities/pull/2869

**Comment by keboliu** [comment]
@StormLiangMS this PR is ready to be merged, would you please help?

**Review by vaibhavhd — CHANGES_REQUESTED** [review]
The version on 202211 and 202205 is same, this needs to be corrected. Blocking merge of this change until that.

File: scripts/db_migrator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vaibhavhd) [reply scripts/db_migrator.py:915]: Why is this resolved?

**Comment by vadymhlushko-mlnx** [comment]
> The version on 202211 and 202205 is same, this needs to be corrected. Blocking merge of this change until that.

I've changed the version to `version_4_0_3` which is the same as for `master`. Is it correct, if not could you please explain how to determine the correct version?

**Comment by vadymhlushko-mlnx** [comment]
> The version on 202211 and 202205 is same, this needs to be corrected. Blocking merge of this change until that.

I've added a placeholder for the `version_3_0_7`. Could you please take a look?

**Comment by vadymhlushko-mlnx** [comment]
> > The version on 202211 and 202205 is same, this needs to be corrected. Blocking merge of this change until that.
> 
> I've added a placeholder for the `version_3_0_7` which is the same as for the `202205` branch. Could you please tell me if it is correct, if no could you please explain the logic behind defining proper DB migration version per branch?

@vaibhavhd kind reminder

**Review by vaibhavhd — APPROVED** [review]

**Pull request merged**